### PR TITLE
Regenerate Search on 2020-06-30-Preview Swagger

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR regenerates `azure-search-documents` using the `2020-06-30-Preview` service version. Not all new features offered in the preview version are being exposed in this change. The following are all the preview service features:

- Semantic search
- Normalizers
- ADLS data sources
- New skillsets

Of these semantic search isn't being exposed.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/CustomEntityLookupSkill.java [hunk 1]
```diff
@@ -21,7 +20,7 @@
 @JsonTypeName("#Microsoft.Skills.Text.CustomEntityLookupSkill")
 @JsonFlatten
 @Fluent
-public class CustomEntityLookupSkill extends SearchIndexerSkill {
+public final class CustomEntityLookupSkill extends SearchIndexerSkill {
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/CustomEntityLookupSkill.java:23]: This was moved from implementation, so adding `final` here is allowed

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/SearchIndexClientImplBuilder.java [hunk 2]
```diff
@@ -208,7 +208,7 @@ public SearchIndexClientImplBuilder addPolicy(HttpPipelinePolicy customPolicy) {
      */
     public SearchIndexClientImpl buildClient() {
         if (apiVersion == null) {
-            this.apiVersion = "2020-06-30";
+            this.apiVersion = "2020-06-30-Preview";
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/SearchIndexClientImplBuilder.java:211]: Can we use the SearchServiceVersion.getLatest()

Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/SearchIndexClientImplBuilder.java:211]: The generated code in-lines this. It is a good question on whether code generation could leverage creating the service enum though.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/models/QueryLanguage.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    /** Static value en-us for QueryLanguage. */
+    public static final QueryLanguage EN_US = fromString("en-us");
+
+    /**
+     * Creates or finds a QueryLanguage from its string representation.
+     *
+     * @param name a name to look for.
+     * @return the corresponding QueryLanguage.
+     */
+    @JsonCreator
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/models/QueryLanguage.java:27]: Do we need @JsonCreator annotation?

Comment (mssfang) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/models/QueryLanguage.java:27]: nvm. It is the implementation file

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ConditionalSkill.java [hunk 4]
```diff
@@ -1,30 +1,35 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ConditionalSkill.java:3]: Should we remove this comment

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/EntityRecognitionSkill.java [hunk 5]
```diff
@@ -1,22 +1,24 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/EntityRecognitionSkill.java:3]: Same here.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ImageAnalysisSkill.java [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
-import java.util.Arrays;
 import java.util.List;
 
-/**
- * A skill that analyzes image files. It extracts a rich set of visual features
- * based on the image content.
- */
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type")
+/** A skill that analyzes image files. It extracts a rich set of visual features based on the image content. */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata\\.type")
```
Comment (mssfang) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ImageAnalysisSkill.java:19]: is "@odata\\.type" intented

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ConditionalSkill.java (hunk 4 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ConditionalSkill.java:3]: Nope, this class is now code generated

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/EntityRecognitionSkill.java (hunk 5 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/EntityRecognitionSkill.java:3]: Nope, this class is now code generated

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ImageAnalysisSkill.java (hunk 6 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/ImageAnalysisSkill.java:19]: Unfortunately, this is a case where code generation is a little over eager on using `JsonFlatten`. When `JsonFlatten` is applied to a class all `JsonProperty` identifiers using a `.` will be treated as getter like calls, so `odata.type` would expected the following JSON:

```json
"odata": {
    "type": "somvalue"
}
```

But in reality the JSON is being returned as `"odata.type": "somevalue"`.

**Review by mssfang — APPROVED** [review]

**Pull request merged**